### PR TITLE
Add support for showing App version and boot status on Network Bootloader

### PR DESCRIPTION
**Pull request by bugobliterator**
image

### CODE REVIEW

File: libraries/AP_CheckFirmware/AP_CheckFirmwareDefine.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#ifndef FORCE_VERSION_H_INCLUDE
+#error AP_CheckFirmwareDefines.h should never be included directly. You probably want to include AP_CheckFirmware/AP_CheckFirmware.h
+#endif
+#include "AP_CheckFirmware.h"
+
+/*
+  declare constant app_descriptor in flash
+ */
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+extern const app_descriptor_t app_descriptor;
```
[the remark this answers is not in the document]
Comment (bugobliterator) [reply libraries/AP_CheckFirmware/AP_CheckFirmwareDefine.h:10]: Done

File: Tools/ardupilotwaf/ap_library.py [hunk 2]
```diff
@@ -171,6 +171,7 @@ class ap_library_check_headers(Task.Task):
     whitelist = (
         'libraries/AP_Vehicle/AP_Vehicle_Type.h',
         'libraries/AP_Common/AP_FWVersionDefine.h',
+        'libraries/AP_CheckFirmware/AP_CheckFirmwareDefine.h'
```
Comment (tridge) [inline Tools/ardupilotwaf/ap_library.py:174]: is this needed?

File: libraries/AP_CheckFirmware/AP_CheckFirmware.cpp [hunk 3]
```diff
@@ -78,42 +81,42 @@ static check_fw_result_t check_firmware_signature(const app_descriptor_signed *a
 /*
   check firmware CRC and board ID to see if it matches
  */
-static check_fw_result_t check_good_firmware_signed(void)
+static check_fw_result_t check_good_firmware_signed()
```
Comment (tridge) [inline libraries/AP_CheckFirmware/AP_CheckFirmware.cpp:84]: revert?

File: Tools/AP_Bootloader/AP_Bootloader.cpp [hunk 4]
```diff
@@ -134,6 +134,21 @@ int main(void)
         try_boot = false;
         led_set(LED_BAD_FW);
     }
+#if AP_BOOTLOADER_NETWORK_ENABLED
+    if (ok == check_fw_result_t::CHECK_FW_OK) {
```
Comment (tridge) [inline Tools/AP_Bootloader/AP_Bootloader.cpp:138]: i'd prefer the memmem here or via a helper function, and not have the global variable

File: Tools/AP_Periph/version.h [hunk 5]
```diff
@@ -5,14 +5,16 @@
 #endif
 
 #include "ap_version.h"
+#include <AP_HAL/AP_HAL.h>
 
 #define THISFIRMWARE "AP_Periph V1.7.0-dev"
 
 // defines needed due to lack of GCS includes
+#ifndef MAVLINK_H
```
Comment (tridge) [inline Tools/AP_Periph/version.h:13]: use HAVE_ENUM_FIRMWARE_VERSION_TYPE

**Review by tridge — CHANGES_REQUESTED** [review]
need testing with signed fw and signed bl

File: Tools/ardupilotwaf/ap_library.py (hunk 2 above)
Comment (bugobliterator) [reply Tools/ardupilotwaf/ap_library.py:174]: fixed, not needed

File: Tools/AP_Bootloader/AP_Bootloader.cpp (hunk 4 above)
Comment (bugobliterator) [reply Tools/AP_Bootloader/AP_Bootloader.cpp:138]: done

File: Tools/AP_Periph/version.h (hunk 5 above)
Comment (bugobliterator) [reply Tools/AP_Periph/version.h:13]: fixed

File: libraries/AP_CheckFirmware/AP_CheckFirmware.cpp (hunk 3 above)
Comment (bugobliterator) [reply libraries/AP_CheckFirmware/AP_CheckFirmware.cpp]: fixed

**Comment by bugobliterator** [comment]
> need testing with signed fw and signed bl

Tested working.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
looks good, will discuss testing on EU dev call

**Pull request merged**